spring-vault-core/src/main/java/org/springframework/vault/authentication/AwsIamAuthenticationOptions.java

```diff
@@ -282,16 +294,27 @@ public AwsIamAuthenticationOptionsBuilder endpointUri(URI endpointUri) {
 			return this;
 		}
 
+		public AwsIamAuthenticationOptionsBuilder useGlobalEndpoint(Boolean useGlobalEndpoint) {
+
+			Assert.notNull(useGlobalEndpoint, "Flag useGlobalEndpoint must not be null");
+
+			this.useGlobalEndpoint = useGlobalEndpoint;
+			return this;
+		}
+
 		/**
 		 * Build a new {@link AwsIamAuthenticationOptions} instance.
 		 * @return a new {@link AwsIamAuthenticationOptions}.
 		 */
 		public AwsIamAuthenticationOptions build() {
 
 			Assert.state(this.credentialsProvider != null, "Credentials or CredentialProvider must not be null");
+			if (useGlobalEndpoint) {
```

Comment:
If using the global endpoint is merely a matter of providing the right `Region`, why don't you just call `regionProvider(() -> Region.US_EAST_1)`?

Reply:
mmh good point you raised here :sweat_smile: . By looking more in details, we are blocked because we are using [spring-cloud-vault-config](https://github.com/spring-cloud/spring-cloud-vault/blob/3.1.x/spring-cloud-vault-config/src/main/java/org/springframework/cloud/vault/config/ClientAuthenticationFactory.java#L276) to instantiate the AwsIamAuthentication, and it does not offer a way to configure the region.
I will create an issue and a PR there. Sorry for that.